src/core/Expector.js

```diff
@@ -84,7 +84,7 @@ getJasmineRequireObj().Expector = function(j$) {
       matcherName: this.matcherName,
       passed: result.pass,
       message: message,
-      error: errorForStack ? undefined : result.error,
+      error: result.error,
```

Comment:
I'm not sure if this should change but I thought it would be better if async matchers were as close to regular matchers as possible

Reply:
This difference is intentional so the stack trace that is displayed to the user includes the lines of code to the test and expectation. The `error` here _should_ be passed in as undefined in order for Jasmine's `ExpectationResult` to generate the correct stack trace.

Reply:
What about when the matcher resolves with an `error` property? Should `ExpectationResult` generate the stack for the returned `error` or the `errorForStack`?

Reply:
If I leave it the way it was the `error` returned by the matcher is just ignored.

Reply:
Should the `error` returned by the matcher just be ignored in async matchers?